### PR TITLE
Spark 3.4

### PR DESCRIPTION
**Pull request by thomas-maschler**
Add support for Spark 3.2.4, 3.3.2 and 3.4.0

### CODE REVIEW

**Comment by pomadchin** [comment]
I'll also publish https://github.com/locationtech/geotrellis/pull/3513 once CQs are cleared!

**Comment by thomas-maschler** [comment]
@pomadchin Rasterframes is still on geotrellis 3.6.3. There were some dependency conflicts which prevented us from updating. I will see if it builds with the latest release

**Comment by pomadchin** [comment]
@thomas-maschler oh, I guess it's stac4s?

**Comment by thomas-maschler** [comment]
yes, I get

```
[error] 	* org.typelevel:cats-effect_2.12:3.4.8 (early-semver) is selected over {3.4.6, 2.5.4}
[error] 	    +- org.locationtech.geotrellis:geotrellis-store_2.12:3.7.0 (depends on 3.4.8)
[error] 	    +- com.softwaremill.sttp.client3:catsce2_2.12:3.7.0   (depends on 2.5.4)
[error] 	    +- co.fs2:fs2-core_2.12:3.6.1                         (depends on 3.4.6)
[error]
[error] 	* co.fs2:fs2-core_2.12:3.6.1 (early-semver) is selected over 2.5.11
[error] 	    +- co.fs2:fs2-io_2.12:3.6.1                           (depends on 3.6.1)
[error] 	    +- org.locationtech.geotrellis:geotrellis-store_2.12:3.7.0 (depends on 3.6.1)
[error] 	    +- com.azavea.stac4s:client_2.12:0.8.1                (depends on 2.5.11)
```

File: project/RFAssemblyPlugin.scala [hunk 1]
```diff
@@ -42,6 +42,13 @@ object RFAssemblyPlugin extends AutoPlugin {
     )
   }
 
+  def chooseJarName(name: String, ver: String, sparkVer: String): String = {
+    if (System.getenv("CI") == null)
+      s"$name-assembly-$sparkVer.jar"
+    else
+      s"$name-assembly-$sparkVer-$ver.jar"
```
Comment (thomas-maschler) [inline project/RFAssemblyPlugin.scala:49]: Do I even need to bother about the spark version in the assembly Jar name?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -25,22 +32,30 @@ jobs:
...
         run: make compile-scala
 
       - name: Test Scala Project
         # python/* branches are not supposed to change scala code, trust them
         if: ${{ !startsWith(github.event.inputs.from_branch, 'python/') }}
-        run: make test-scala
+        env:
+          SPARK_VERSION: ${{ matrix.spark_version }}
+        run:
+          ulimit -c unlimited
```
Comment (thomas-maschler) [inline .github/workflows/ci.yml:45]: @pomadchin, without this, test suite started to crash for Spark versions <3.4 after I added frameless 0.14.1. Any idea why?

**Comment by thomas-maschler** [comment]
@echeipesh, the publishing CI workflow is still incomplete and only builds jar and wheel on the worker but does not publish the files to Maven or Pypi. Do you want to keep this manual for now?

File: .github/workflows/ci.yml (hunk 2 above)
Comment (pomadchin) [reply .github/workflows/ci.yml:45]: Nope, that's a good observation though

**Comment by pomadchin** [comment]
@thomas-maschler stac4s migrated into the stac org! https://github.com/stac-utils/stac4s 

I'll update it once I have some time (:

**Comment by pomadchin** [comment]
🔥

**Comment by pomadchin** [comment]
Let's merge it as is for now!

**Pull request merged**